### PR TITLE
updated documentation for scales and data

### PR DESCRIPTION
**Pull request by jckr**
Documentation - the final frontier! 
After writing on colors, I've taken a stab at scales and data.

### CODE REVIEW

File: docs/markdown/scales-and-data.md [hunk 1]
```diff
@@ -46,3 +120,15 @@ All scale properties are prepended with the name of the attribute, for instance
...
+
+Scales can be defined either at the XYPlot level, in which case they apply to the whole chart, or at the series level. Scales defined at the series level override those defined at the XYPlot level. 
+
+The scales provided to each individual series don't have to have the same parameters. For instance, if you wanted to do a dual-axis chart, you could provide a different yDomain and yRange to two data series (use at your own risk).
+
+### Other uses of scales
+
+Scales can also be used in [Axes](axes.md) and in [Gridlines](grids.md). You can pass an x-scale (so xDomain, xRange, xPadding, xType) to an XAxis or a VerticalGridLines component, and a y-scale (so yDomain, yRange, yPadding, yType) to a YAxis or HorizontalGridLines component. These scale parameters don't have to be the same as the ones passed to your series. 
+
+Did you guess that this data set was the digits sorted alphabetically? eight, five, four...
```
Comment (mcnuttandrew) [inline docs/markdown/scales-and-data.md]: 😂😂😂

File: showcase/index.js [hunk 2]
```diff
@@ -63,6 +63,10 @@ import {
   CustomPalette
 } from './color/mini-color-examples';
 
+import {
```
Comment (mcnuttandrew) [inline showcase/index.js]: Maybe batch the color and data folders into a small multiples folder?

File: showcase/data/mini-data-examples.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  {x: 9, y: 0}
+];
+
+const defaultProps = {
+  width: 200,
+  height: 200,
+  margin: {top: 5, left: 5, right: 5, bottom: 5}
+};
+
+export function MiniCharts() {
```
Comment (mcnuttandrew) [inline showcase/data/mini-data-examples.js]: Tests plox

File: docs/markdown/interaction.md [hunk 4]
```diff
@@ -0,0 +1 @@
+## Interaction
```
Comment (mcnuttandrew) [inline docs/markdown/interaction.md]: ?

**Comment by jckr** [comment]
Adding more documentation. This includes:
*  a new "interactions" page which covers all interaction handlers plus strategies we're using to implement interactions in our charts, 
* new AreaSeries and RectSeries pages, which didn't exist previously, and both series are different enough from LineSeries and BarSeries to deserve their own page, and
* a complete pass on all series page for consistency, correction and completeness. 
screenshot 2017-06-27 13 13 37
screenshot 2017-06-27 13 13 20
screenshot 2017-06-27 13 13 03
screenshot 2017-06-27 13 12 35

**Pull request merged**